text/0000-tracked-properties.md

```diff
@@ -0,0 +1,1029 @@
+
+### Leverage Existing JavaScript Knowledge
+
+Ember's computed properties provide functionality that overlaps with native
+JavaScript getters and setters. Because native getters don't provide Ember with
+the information it needs to track changes, it's not possible to use them
+reliably in templates or in other computed properties.
+
+New learners have to "unlearn" native getters, replacing them with Ember's
+computed property system. Unfortunately, this knowledge is not portable to other
+applications that don't use Ember that developers may work on in the future, and
+while this problem may be lessened by adopting native classes and decorators, it
+still requires users learn Ember's notification system and its quirks.
+
+Tracked properties are as thin a layer as possible on top of native JavaScript.
+Tracked properties look like normal properties because they _are_ normal
+properties.
+
+Because there is no special syntax for retrieving a tracked property, any
+JavaScript syntax that feels like it should work does work:
+
+```js
+// Dot notation
+const fullName = person.fullName;
+// Destructuring
+const { fullName } = person;
+// Bracket notation for computed property names
+const fullName = person['fullName'];
+```
+
+Similarly, syntax for changing properties works just as well:
+
+```js
+// Simple assignment
+this.firstName = 'Yehuda';
+// Addition assignment (+=)
+this.lastName += 'Katz';
+// Increment operator
+this.age++;
+```
+
+This compares favorably with APIs from other libraries, which becomes more
+verbose than necessary when JavaScript syntax isn't available:
+
+```js
+this.setState({
+  age: this.state.age + 1,
+});
+```
+
+```js
+this.setState({
+  lastName: this.state.lastName + "Katz";
+})
+```
+
+### Avoiding Dependency Hell
+
+Currently, Ember requires developers to manually enumerate a computed property's
+dependent keys: the list of _other_ properties that _this_ computed property
+depends on. Whenever one of the listed properties changes, the computed
+property's cache is cleared and any listeners are notified that the computed
+property has changed.
+
+In this example, `'firstName'` and `'lastName'` are the dependent keys of the
+`fullName` computed property:
+
+```js
+import EmberObject, { computed } from '@ember/object';
+
+const Person = EmberObject.extend({
+  fullName: computed('firstName', 'lastName', function() {
+    return `${this.firstName} ${this.lastName}`;
+  }),
+});
+```
+
+While this system typically works well, it comes with its share of drawbacks.
+
+First, it's annoying to have to type every property twice: once as a string as a
+dependent key, and again as a property lookup inside the function. While
+explicit APIs can often lead to clearer code, this verbosity often obfuscates
+the intent of the property. People understand intuitively that they are typing
+out dependent keys to help _Ember_, not other programmers.
+
+Second, people tell us that this syntax is not very intuitive. You have to read
```

Comment:
I'm not sure where this comes from, I can vouch for it anecdotally but I don't have specific examples to link to. Should we remove it from the RFC altogether?

Reply:
I would prefer avoiding this kind of statement unless it can be backed up. Otherwise, it just sounds like a [weasel word](https://en.wikipedia.org/wiki/Weasel_word).